### PR TITLE
Add Freshdesk connector

### PR DESCRIPTION
**Pull request by kapilraajP**
# Description

Adding Freshdesk connector

Fixes: https://github.com/ballerina-platform/ballerina-extended-library/issues/362

One line release note: 
- Add Freshdesk connector

## Type of change

- [X] New feature (non-breaking change which adds functionality)

**Test Configuration**:
* Ballerina Version: Ballerina Swan Lake 2201.0.2
* Operating System: Linux 
* Java SDK: Java 11

# Checklist:

### Changes to OpenAPI definition

- [X] Modified `info` section of the OpenAPI definition file - [Example](https://github.com/ballerina-platform/ballerina-extended-library/discussions/74)

    - `description` - what the connector is about 
    - `x-ballerina-init-description` OpenAPI extension - connector initialization details 
    - `x-ballerina-display` OpenAPI extension - connector name and path to the connector icon 

- [X] Added proper description each API Key, if API Key authentication is defined in OpenAPI definition file 

### Changes to connector module

- [X] Added license header at the top of each .bal file. 
- [X] Added Package.md as per the guide [here](https://github.com/ballerina-platform/ballerina-extended-library/discussions/77)
- [X] Added Module.md for the module of the  connector as per the guide [here](https://github.com/ballerina-platform/ballerina-extended-library/discussions/78)
- [ ] Added `icon.png`
- [X] Added Ballerina.toml file with following information. For keywords refer to [guide](https://github.com/ballerina-platform/ballerina-extended-library/discussions/72)

### Verifying connector module

- [X] Compiled the connector successfully with the targeted ballerina version. 
- [X] Conducted smoke tests on the connector (Recommended when OpenAPI definition is obtained from an unofficial source)

### Security checks
 - [X] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? 
 - [X] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: openapi/freshdesk/Dependencies.toml [hunk 1]
```diff
@@ -0,0 +1,299 @@
...
+]
+
+[[package]]
+org = "ballerinax"
+name = "freshdesk"
+version = "1.0.0"
+dependencies = [
+	{org = "ballerina", name = "http"},
+	{org = "ballerina", name = "io"},
+	{org = "ballerina", name = "test"},
```
Comment (aneeshafedo) [inline openapi/freshdesk/Dependencies.toml:292]:
```suggestion
\```

File: openapi/freshdesk/Module.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+```ballerina
+import ballerinax/freshdesk;
+```
+
+### Step 2: Create a new connector instance
+Create a `freshdesk:ClientConfig` with the api key as username in basic authentication password you can provide any, and initialize the connector with it.
+```ballerina
+configurable http:CredentialsConfig & readonly authConfig = ?;
+freshdesk:ClientConfig clientConfig = {auth : authConfig};
+freshdesk:Client baseClient = check new Client(clientConfig,
```
Comment (aneeshafedo) [inline openapi/freshdesk/Module.md:25]:
```suggestion
freshdesk:Client baseClient = check new (clientConfig, 
```

File: openapi/freshdesk/Module.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+Freshdesk is a cloud-based customer service software that enables businesses of all sizes to deliver stellar customer support for requests coming in via email, web, phone, chat, and social into tickets, and unifies ticket resolution across channels.
+
+## Prerequisites
+Before using this connector in your Ballerina application, complete the following:
+1. Create an [Freshdesk account](https://freshdesk.com/signup). 
+2. Obtain tokens - Follow [this guide](https://developers.freshdesk.com/api/#getting-started).
+
+## Quickstart
+
+To use the Freshdesk connector in your Ballerina application, update the .bal file as follows
```
Comment (LakshanSS) [inline openapi/freshdesk/Module.md:12]:
```suggestion
## Quickstart
To use the Freshdesk connector in your Ballerina application, update the .bal file as follows
```

File: openapi/freshdesk/Ballerina.toml [hunk 4]
```diff
@@ -0,0 +1,12 @@
+[package]
+license = ["Apache-2.0"]
+keywords = ["Sales & CRM/Customer Relationship Management", "Cost/Freemium"]
+org = "ballerinax"
+name = "freshdesk"
+distribution = "2201.0.2"
+repository = "https://github.com/ballerina-platform/ballerinax-openapi-connectors"
+version = "1.0.0"
+authors = ["Ballerina"]
```
Comment (LakshanSS) [inline openapi/freshdesk/Ballerina.toml:9]: icon has to be added after its available

File: openapi/freshdesk/Package.md [hunk 5]
```diff
@@ -0,0 +1,19 @@
+Connects to [Freshdesk](https://developers.freshdesk.com/api/#intro) from Ballerina
+
+## Package overview
+The `ballerinax/Freshdesk` is a [Ballerina](https://ballerina.io/) connector for Freshdesk API.
```
Comment (SkNuwanTissera) [inline openapi/freshdesk/Package.md:4]:
```suggestion
The `ballerinax/freshdesk` is a [Ballerina](https://ballerina.io/) connector for Freshdesk API.
```

**Review by SkNuwanTissera — APPROVED** [review]

**Review by aneeshafedo — APPROVED** [review]

**Pull request merged**